script.js

```diff
@@ -0,0 +1,170 @@
+          <h3>${recipe.title}</h3>
+          <div class='divider'></div> 
+          <p><strong>Cuisine:</strong> 
+            <span class='cuisine-value'>
+              ${recipe?.cuisines?.length ? recipe.cuisines.join(', ') : 'N/A'}
+            </span>
+          </p>
+          <p><strong>Time:</strong> 
+            <span class='time-value'>${recipe.readyInMinutes} min</span>
+          </p>
+          <div class='divider'></div>   
+          <p class='ingredients-header'>Ingredients:</p>
+          <ul>
+            ${recipe?.extendedIngredients
+              .map(({ name }) => `<li>${name}</li>`)
+              .join('')}
+          </ul>
+        </div>
+      </div>
+    `
+  })
+}
+
+const fetchData = async () => {
+  const cachedData = localStorage.getItem(CACHE_KEY)
+
+  if (cachedData) {
+    try {
+      const { recipes, timestamp } = JSON.parse(cachedData)
+      const isExpired = Date.now() - timestamp > CACHE_DURATION
+      recipesArray = recipes
+
+      if (!isExpired) {
+        showCardsContainer(recipes)
+        return
+      }
+    } catch (error) {
+      console.error('Error parsing cached data:', error)
+      localStorage.removeItem(CACHE_KEY)
+    }
+  }
+
+  try {
+    const res = await fetch(URL)
+
+    if (res.status === 402) {
+      cardsContainer.innerHTML = `
+        <div class='empty-message'>
+          <h2>Daily API limit reached 🚫</h2>
+          <p>Please try again later.</p>
+        </div>`;
+      return; 
+    }
+
+    if (!res.ok) {
+      cardsContainer.innerHTML = `
+        <div class='empty-message'>
+          <h2>Oops 😅</h2>
+          <p>No recipes found...</p>
+        </div>  
+      `
+      throw new Error(`HTTP error! Status: ${res.status}`)
+    }
+
+    const data = await res.json()
+    const recipes = data.results
+    recipesArray = recipes
+
+    localStorage.setItem(
+      CACHE_KEY,
+      JSON.stringify({
+        recipes,
+        timestamp: Date.now()
+      })
+    )
+
+    showCardsContainer(recipes)
+  } catch (error) {
+    console.error('Error occurred:', error)
+  }
+}
+
+fetchData()
+
+const filterOnCuisine = () => {
+  const selected = cuisineFilter.querySelector('input[name="cuisine"]:checked')
+  const selectedCuisine = selected ? selected.value : 'all'
+
+  const filteredRecipes =
+    selectedCuisine === 'all'
+      ? recipesArray
+      : recipesArray.filter((recipe) =>
+          recipe.cuisines
+            .map((cuisine) => cuisine.toLowerCase())
+            .includes(selectedCuisine.toLowerCase())
+        )
+
+  showCardsContainer(filteredRecipes)
+}
+
+const sortOnTime = () => {
+  const selected = sortSection.querySelector('input[name="order"]:checked')
+  const order = selected ? selected.value : 'desc'
+
+  const sorted = [...recipesArray].sort((a, b) => {
+    return order === 'asc'
+      ? a.readyInMinutes - b.readyInMinutes
+      : b.readyInMinutes - a.readyInMinutes
+  })
+
+  showCardsContainer(sorted)
+}
```

Comment:
nice job! If you want to keep the sorting even when you change the cuisine filters you could maybe save the sorting rule in a global variable. Just something to take with you going forward.